docs/readmes/resources/ref_magma_metrics.md

```diff
@@ -0,0 +1,161 @@
+
+Similar to Python services, _magmad_ divides the metrics collected from C++ services into chunks of 1 MB or less, and uploads them over gRPC to _metricsd_ in the Orc8r.
+
+## Phase 2: Orchestrator and Prometheus
+
+The _metricsd_ cloud service reads metric names and labels from enums defined in [`metricsd.proto`](https://sourcegraph.com/github.com/magma/magma@v1.6.0/-/blob/orc8r/protos/metricsd.proto) and accepts gRPC messages accordingly. Once it has received metrics from the AGW, _metricsd_ pushes them again over gRPC to the [`MetricsExporter`](https://sourcegraph.com/github.com/magma/magma@v1.6.0/-/blob/orc8r/cloud/go/services/metricsd/protos/exporter.proto) servicers in its `Collect` method.
+
+```go
+// metricsd/servicers/servicer.go
+
+func (srv *MetricsControllerServer) Push(ctx context.Context, in *protos.PushedMetricsContainer) (*protos.Void, error) {
+    // ...
+    metricsExporters, err := metricsd.GetMetricsExporters()
+    // ...
+    for _, e := range metricsExporters {
+        err := e.Submit(metricsToSubmit)
+        // ...
+    }
+```
+
+This is one of the [extensions supported in the Orc8r](orc8r/architecture_modularity) and the label `orc8r.io/metrics_exporter` can be used to export metrics to any data sink. By default, the `GRPCPushExporterServicer`, which is registered in the [`orchestrator`](https://sourcegraph.com/github.com/magma/magma@v1.6.0/-/blob/orc8r/cloud/go/services/orchestrator/orchestrator/main.go?L66) service, is the only metrics exporter. So `GRPCPushExporterServicer`, through its `Submit` method, pushes metrics over gRPC to Prometheus Edge Hub.
+
+```go
+// grpc_exporter_servicer.go
+
+func (s *GRPCPushExporterServicer) pushFamilies(families []*io_prometheus_client.MetricFamily) error {
+    // ...
+    client, err := s.getClient()
+    if err != nil {
+        return err
+    }
+    _, err = client.Collect(context.Background(), &edge_hub.MetricFamilies{Families: families})
+    // ...
+}
+```
+
+[Prometheus Edge Hub](https://github.com/facebookincubator/prometheus-edge-hub) is a Facebook project that replaces the Prometheus Pushgateway. Orc8r's Prometheus service scrapes and drains the Edge Hub so metrics finally arrive at their home in the Prometheus server. On a dev environment, the Prometheus server runs at [localhost:9090](http://localhost:9090).
```

Comment:
1. Add link to the Prometheus scrape targets code I previously sent a link to
2. Add a link for "on a dev environment" to the relevant part of the docker-compose.yml that sets this up